ThatValidator.js

```diff
@@ -362,10 +347,10 @@
 
                 for(var i = 0; i < elements.length; i++)
                 {
-                    var field = elements[i];
+                    var field2 = elements[i];
 
                     //if the element has not already been found
-                    if(self.fields.indexOf(field) == -1)
+                    if(self.fields.indexOf(field2) == -1)
```

Comment:
I think you did it in the wrong place then. I see what you're talking about in the validate() function, but this is in the init() function